src/lib.rs

```diff
@@ -361,6 +365,27 @@ impl Consul {
         self.read_key(req).await
     }
 
+    /// Returns the services currently registered with consul for the datacenter of the agent being queried.
+    /// See https://www.consul.io/api-docs/catalog#list-services for more information.
+    /// # Arguments:
+    /// - timeout: The timeout [`Duration`](Duration) to apply to the request.
+    /// # Errors:
+    /// [ConsulError](consul::ConsulError) describes all possible errors returned by this api.
+    pub async fn get_all_registered_service_names(
+        &self,
+        timeout: Option<Duration>,
```

Comment:
The other methods do not have a timeout. Could we add this timeout at the client level? I don't think that having some methods that accept a timeout & some that do not is great for consumers.

Reply:
I think having timeouts at the specific method level is probably helpful for use cases where a global default will not work. Adding a default to the client is a good idea, I can do that, but in a separate pr.

Reply:
added